funfact/model/_factorization.py

```diff
@@ -76,15 +80,52 @@ def forward(self):
         '''Evaluate the tensor expression the result.'''
         return self.tsrex | self._evaluator
 
-    def _get_elements(self, idx):
+    def _get_elements(self, key):
         '''Get elements at index of tensor expression.'''
-        new_idx = []
-        for i in idx:
+
+        # Generate full index list
+        full_idx = []
+        ellipsis_i = None
+        for i in key:
             if isinstance(i, int):
-                new_idx.append(slice(i, i+1))
+                if i != -1:
+                    full_idx.append(slice(i, i+1))
+                else:
+                    full_idx.append(slice(i, None))
+            elif isinstance(i, slice):
+                full_idx.append(i)
+            elif i is Ellipsis:
+                ellipsis_i = key.index(...)
             else:
-                new_idx.append(i)
-        _index_slicer = SlicingPropagator(new_idx)
+                raise IndexError(
+                    f'Unrecognized index {i} of type {type(i)}'
+                )
+        if ellipsis_i is not None:
```

Comment:
Could this handle the case where `...` appears at the end or middle of the indices?

Reply:
Yes, normally it should because of the `.insert`. It is assumed however that there is at most one `...`. I think same assumption as numpy makes, otherwise you can't infer the sizes.